src/main/kotlin/com/grepp/nbe1_3_team04/vote/domain/Vote.kt

```diff
@@ -0,0 +1,80 @@
+package com.grepp.nbe1_3_team04.vote.domain
+
+import com.grepp.nbe1_3_team04.global.domain.BaseEntity
+import jakarta.persistence.*
+import jakarta.validation.constraints.NotNull
+import org.hibernate.annotations.SQLDelete
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+@Table(
+    indexes = [Index(name = "idx_vote_team_id", columnList = "teamId"), Index(
+        name = "idx_vote_member_id",
+        columnList = "memberId"
+    )]
+)
+@SQLDelete(sql = "UPDATE vote SET is_deleted = 'TRUE' WHERE vote_id = ?")
+@Entity
+class Vote private constructor(
+    val memberId: Long,
+    val teamId: Long,
+    title: String,
+    endAt: LocalDateTime
+) : BaseEntity() {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    val voteId: Long? = null
+
+    @Column(length = 50)
+    var title: @NotNull String? = null
```

Comment:
```suggestion
    var title: @field:NotNull String? = null
```

해당 부분이 엔티티에서는 어떻게 작동하는지 정확하게는 모르는데 @filed 를 선언해주지 않으면 기본값으로 코틀린에서 @get:NotNull 이 들어간다고 하더라구요.
붙는 어노테이션에 따라 아래와 같이 검증을 해준다고 합니다!

file: 파일
property: Kotlin 의 property (이 use site 를 사용하면 Java에선 볼 수 없음)
field: Java 의 field(Kotlin 의 property)
get: property getter
set: property setter
receiver: 확장 함수나 속성의 수신 객체 getter
param: constructor getter
setparam: property setter getter
delegate: delegate property 의 인스턴스를 저장하는 field

Reply:
아... 무슨 말인지 이해를 못하긴했는데 반영하겠습니다.. 어노테이션의 지정범위를 말하는걸까요 
근데 제가 궁금한건 ?가 붙여서 Null일 수 있어 해놓고 어노테이션으로 NotNull이다 이게 무슨..?
실제로 Java에서는 Null이 될 수 있기 때문에 막아주지만 이런 경우에는 어떻게 해야될까요 ? 그냥 NotNull 쓰던 이유가 nullable = false를 위해서였으니까 바꿔야될까요 ? 컬럼으로 변경해야될까요 ?

Reply:
근데 제가 발표했던 내용이 RequestDto 관련 내용이었어서 Entity의 경우에 null 값이 들어가면 안되는 경우 그냥 ?로 선언을 안해주시면 될 것 같습니다!

Dto 에서 @Valid 를 통해 검증 후 원하는 오류 메시지를 뿌려주려면 nullable 하게 선언을 해주고 @filed:NotNull(message =" 원하는 오류 메시지") 를 작성해주었거든요!

Reply:
아하 감사합니다!! 나중에 DTO에 검증하는 부분은 재경님 말씀처럼 해봐야겠네요 ! 나중에 다시 참고하도록 하겠습니다 지금은 그냥 ?를 지우는 걸로 하겠습니다 !